src/js/components/ClickToSelect.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+const METHODS_TO_BIND = [
+  'selectAll'
+];
+
+class ClickToSelect extends React.Component {
+  constructor() {
+    super();
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  selectAll(event) {
+    event.preventDefault;
+    let selection = document.getSelection();
+    let node = this.target;
+
+    selection.selectAllChildren(node);
+  }
+
+  render() {
+    return (
+      <span
+        onClick={this.selectAll}
+        ref={(node) => this.target = node}>
```

Comment:
We could just pass a string to `ref` and then use that in place of `this.target`. So you would reference the DOM node by using `this.refs.{someString}`